Tests/Tests/STPBankAccountTest.m

```diff
@@ -100,53 +117,82 @@ - (void)testBankAccountEquals {
     STPBankAccount *bankAccount1 = [STPBankAccount decodedObjectFromAPIResponse:[self completeAttributeDictionary]];
     STPBankAccount *bankAccount2 = [STPBankAccount decodedObjectFromAPIResponse:[self completeAttributeDictionary]];
 
-    XCTAssertEqualObjects(bankAccount1, bankAccount1, @"bank account should equal itself");
-    XCTAssertEqualObjects(bankAccount1, bankAccount2, @"bank account with equal data should be equal");
+    XCTAssertNotEqual(bankAccount1, bankAccount2);
+
+    XCTAssertEqualObjects(bankAccount1, bankAccount1);
+    XCTAssertEqualObjects(bankAccount1, bankAccount2);
+
+    XCTAssertEqual(bankAccount1.hash, bankAccount1.hash);
+    XCTAssertEqual(bankAccount1.hash, bankAccount2.hash);
 }
 
 #pragma mark - Description Tests
 
-- (void)testDescriptionWorks {
+- (void)testDescription {
     STPBankAccount *bankAccount = [STPBankAccount decodedObjectFromAPIResponse:[self completeAttributeDictionary]];
-    bankAccount.routingNumber = @"123456789";
     XCTAssert(bankAccount.description);
 }
 
 #pragma mark - STPAPIResponseDecodable Tests
 
 - (NSDictionary *)completeAttributeDictionary {
```

Comment:
Test json like this should probably be moved to STPFixtures? And/or an actual json file.